### ISSUE TITLE
Color does not apply to the first character

### ISSUE THREAD

**Issue by sayanarijit** [uttarayan21/ansi-to-tui#3]
Not sure if it's related to `tui-rs` or this crate, but the color doesn't seem to apply to the first character

![image](https://user-images.githubusercontent.com/11632726/119299470-9499b500-bc7c-11eb-9f10-a7df087f100d.png)

Code: https://github.com/sayanarijit/xplr/blob/cb695fcaa7b550007b19597ed18ec2ba31b25e4a/src/init.lua#L2060 and https://github.com/sayanarijit/xplr/blob/cb695fcaa7b550007b19597ed18ec2ba31b25e4a/src/ui.rs#L553

**Comment by sayanarijit** [comment]
Also, it would be great if the `Error` implements `Display` and supports the `anyhow` crate.

**Comment by uttarayan21** [comment]
Oh I didn't see this issue before publishing the new release.
I'll try to reproduce this.
Probably an issue with the parser logic.

**Comment by uttarayan21** [comment]
Can you post a small reproducible example ? 
I can't seem to reproduce with this.
```rust
#[test]
fn nocode_code() {
    let string = "foo \x1b[33myellow\x1b[31mred\x1b[0m";
    println!("{:#?}",ansi_to_text(string.bytes()));
}
```
Or could you print the actual ansi text and the variable that is causing this issue ?

> Also, it would be great if the Error implements Display and supports the anyhow crate.

I could Implement display but I don't know how one would want to display the error. Like for example someone using the error in a language other than english.
As for the `anyhow` crate I'm already using this crate in some other projects and I'm using a different error crate and would like to keep the footprint minimal, so probably not.

**Comment by sayanarijit** [comment]
> ```rust
> #[test]
> fn nocode_code() {
>     let string = "foo \x1b[33myellow\x1b[31mred\x1b[0m";
>     println!("{:#?}",ansi_to_text(string.bytes()));
> }
> ```

```rust
#[test]
fn nocode_code() {
    let string = "\x1b[33myellow\x1b[31mred\x1b[0m";
    println!("{:#?}",ansi_to_text(string.bytes()));
}
```

produces

```rust
Ok(
    Text {
        lines: [
            Spans(
                [
                    Span {
                        content: "yellow",
                        style: Style {
                            fg: None,
                            bg: None,
                            add_modifier: (empty),
                            sub_modifier: (empty),
                        },
                    },
                    Span {
                        content: "red",
                        style: Style {
                            fg: Some(
                                Red,
                            ),
                            bg: None,
                            add_modifier: (empty),
                            sub_modifier: (empty),
                        },
                    },
                ],
            ),
        ],
    },
)
```

Notice the `fg: None` part.

**Comment by sayanarijit** [comment]
Created https://github.com/uttarayan21/ansi-to-tui/pull/4, I think implementing std error is a standard practice. `anyhow` will support anything that implements this. Users, not satisfied with the default Display, should be able to implement it themselves.

**Comment by sayanarijit** [comment]
Although, it'd definitely help if library provides more error context via enum arguments. Even if plain string.

**Comment by uttarayan21** [comment]
> Created #4, I think implementing std error is a standard practice. `anyhow` will support anything that implements std error. Users, not satisfied with the default Display, should be able to implement it themselves.

Oh I see ! Thanks for the clarification. I didn't know to support anyhow you just need `std::error::StdError`.

**Comment by uttarayan21** [comment]
![xplr](https://user-images.githubusercontent.com/18732700/119677403-51963800-be5c-11eb-8754-33191dc028a6.png)
Seems to work now I'll publish a release once I do some tests.

**Comment by uttarayan21** [comment]
I just published version v0.2.1 which should fix the issue.

**Comment by sayanarijit** [comment]
Thanks!
